react-js/react.md

```diff
@@ -41,29 +43,93 @@ You will spend the day working through these materials.
+
+### Examples
+- Hello World component
+```import React from 'react'
+
+class HelloWorld extends React.Component {
+    render() {
+        return <h1>Hello, World!</h1>
+    }
+}
+
+export default HelloWorld
+```
+
+- Component with a single prop
+```class Parent extends React.Component{
+  doSomething(){
+     console.log("Parent component");
+  }
+  render() {
+    return <div>
+         <Child 
+           title="Title 1" 
+           onClick={this.doSomething} />
+         <Child 
+           title="Title 2" 
+           onClick={this.doSomething} />
+      </div>
+  }
+}
+
+class Child extends React.Component{
+  render() {
+    return <div>
+       <h1>{this.props.title}</h1>
+      </div>
+  }
+}
+```
+
+- Component assigning a `classname`
+```// ParentComponent.js
+import React from 'react';
+import { ChildComponent } from '../child/ChildComponent';
+
+export const ParentComponent = () =>
+  <div className="parent-component">
+    <ChildComponent className="parent-component__child">
+      ...
+    </ChildComponent>
+  </div>
+
+// ChildComponent.js
+import React from 'react';
+
+export const ChildComponent = ({ className, children }) =>
+  <div className={`some-css-className ${className}`}>
+    {children}
+  </div>
+```
+will render as:
+```
+<div class="parent-component">
+  <div class="some-css-className parent-component__child">
+    ...
+  </div>
+</div>
+```
 
 ### Common Mistakes / Misconceptions
 
-1. JSX tag names for components must match up with component names, both in spelling and case.
-2. In render functions, don't forget to return JSX.
-3. Don't forget to specify both arguments in ReactDOM.render function.
-4. Don't forget `this` when working within a React class. Examples: `this.state.thing`
-5. Don't forget that variables in JSX must be enclosed in brackets and don't forget number 4. Example: `<div>Hello {this.state.thing}</div>`
-6. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
-7. The term 'ReactJS' is commonly used to clarify that it's a javascript framework; however, 'React' is its official name.
+1. In render functions, don't forget to return JSX.
+2. Don't forget that variables in JSX must be enclosed in brackets and don't forget number 4. Example: `<div>Hello {this.state.thing}</div>`
+3. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
 
 ### Guided Practice
--[Importing & Exporting with React](https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac)
+- [Importing & Exporting with React](https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac)
+- [Rendering a List using map](https://codepen.io/BigNom/pen/BZZOgX)
```

Comment:
- Use ES6 arrow function in the map function
- Add key property inside the map function
- use `<ol>` instead of `<ul>`
- Do proper indentation of code

Reply:
Already added these points

Reply:
@jamesjose03 I am talking about implementing the above points in the codepen example :
```class ShowList extends React.Component {
  render() {
    return (
    <div>
        <h3> Friends</h3>
        <ul>
        {this.props.names.map(function(friend){
            return <li> {friend} </li>;
          })}
        </ul>
        </div>
    )
  }
}

Reply:
Oh okay. Will do it.

Reply:
Done. Pls check @Aayush-1999